packages/medusa/src/services/product-variant.ts

```diff
@@ -333,6 +317,48 @@ class ProductVariantService extends BaseService {
     })
   }
 
+  async updateVariantPrices(
+    variantId: string,
+    prices: ProductVariantPrice[]
+  ): Promise<void> {
+    return this.atomicPhase_(async (manager: EntityManager) => {
+      const moneyAmountRepo = manager.getCustomRepository(
+        this.moneyAmountRepository_
+      )
+
+      // get prices to be deleted
+      const obsoletePrices = await moneyAmountRepo
+        .createQueryBuilder("money_amount")
+        .where("money_amount.variant_id = :variant_id", {
+          variant_id: variantId,
+        })
+        .andWhere(
+          new Brackets((qb) => {
+            qb.where("money_amount.currency_code NOT IN (:...currency_codes)", {
+              currency_codes: prices.map((p) => p.currency_code),
+            }).orWhere("money_amount.region_id NOT IN (:...region_ids)", {
+              region_ids: prices.map((p) => p.region_id),
+            })
+          })
+        )
+        .getMany()
+
```

Comment:
This is brilliant - one thing I would like is that this gets moved into the `MoneyAmountRepository` as a custom repo method. Generally, we should avoid having query builder stuff in the service layer.

Also I think some of the queries here can be TypeORM-ified more: e.g. you can do:
```ts
qb.where({
  currency_code: Not(In(prices.map(p => p.currency_code)))
})
```

Reply:
Also `.where({ variant_id: variantId })`